appinventor/components/src/com/google/appinventor/components/runtime/util/NativeOpenStreetMapController.java

```diff
@@ -416,6 +419,25 @@ public MapFactory.MapType getMapType() {
     return MapFactory.MapType.values()[tileType.toUnderlyingValue()];
   }
 
+  @Override
+  public String getCustomUrl() {
+    return customUrl;
+  }
+
+  @Override
+  public void setCustomUrl(String url) {
+    if (!url.equals(this.customUrl)) {
+      // Clear any cached tiles if the custom URL changes
+      // TODO: even though isCleared returns true, the cache is still not empty.  Debug further.
+      // Only solution is to use Android's Storage > Clear cache, which does work.
+      SqlTileWriter sqlTileWriter = new SqlTileWriter();
+      boolean isCleared = sqlTileWriter.purgeCache(); //view.getTileProvider().getTileSource().name()); //"Custom"); 
+      // setRotationEnabled(isCleared);
+      view.getTileProvider().clearTileCache();
+    }
+    this.customUrl = url;
```

Comment:
It seems as though after changing the custom URL, if `tileType == Custom` then we need to regenerate the tileSource since customUrl is captured when it is constructed on line 457. Otherwise, the tile source will still be using the previous URL.

Reply:
Noted & fixed.